test/adjoint/test_adjoint_mesh_seq.py

```diff
@@ -369,6 +368,16 @@ def test_num_enrichments_error(self):
         msg = "A positive number of enrichments is required."
         self.assertEqual(str(cm.exception), msg)
 
+    def test_form_error(self):
+        mesh_seq = self.go_mesh_seq(self.get_function_spaces_decorator("R", 0, 0))
+        with self.assertRaises(AttributeError) as cm:
+            mesh_seq.forms()
+        msg = (
+            "Forms have not been read in. Use read_forms({'field_name': F}) in"
+            " get_solver to read in the forms."
+        )
+        self.assertEqual(str(cm.exception), msg)
+
```

Comment:
I added this test and another in the `TestGoalOrientedMeshSeq` class, in this same file below

Reply:
Oh sorry, I haven't! I was sure I did. Thanks for adding it